micro_dl/inference/image_inference.py

```diff
@@ -884,17 +894,37 @@ def run_prediction(self):
                     )
                     pred_fnames.append(pred_fname)
                 if self.metrics_inst is not None:
-                    print('Computing metrics on time {} position {}'.format(time_idx, pos_idx))
                     if not self.mask_metrics:
                         mask_image = None
-
                     self.estimate_metrics(
                         target=target_image[i],
                         prediction=pred_image[i],
                         pred_fnames=pred_fnames,
                         mask=mask_image,
                     )
+
+                with tqdm(total=len(chan_slice_meta['slice_idx'].unique()), desc='z-stack saving', leave=False) as pbar_s:
+                    for j, z_idx in enumerate(chan_slice_meta['slice_idx'].unique()):
+                        im_name = aux_utils.get_im_name(
+                            time_idx=time_idx,
+                            channel_idx=chan_idx,
+                            slice_idx=z_idx,
+                            pos_idx=pos_idx,
+                            ext="",
+                            extra_field="xy0",
+                        )
+                        print(np.shape(input_image), np.shape(input_image[..., j]))
+                        self.save_pred_image(
+                            im_input=input_image[..., j],
+                            im_target=target_image[i][:, :, j],
```

Comment:
I think this might also break for 3D inference but I'm not sure since the tests are broken. :-)

I can fix it later after this branch is merged in my master_tests PR.